File: task_2/hist.py
"""
Tâche 2:
1. Histogramme en Python
"""

# Importons les bibliothèques nécessaires
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

# Chargeons le dataset
df = pd.read_csv("./Housing_cleaned.csv")

# Réalisons l'hstogramme

plt.figure(figsize=(10, 6))

plt.hist(df['bedrooms'], bins=10, color='lightblue', edgecolor='black', alpha=0.7)

plt.xlabel('Nombre de chambres\n', fontsize=12)
plt.ylabel('Fréquence\n', fontsize=12)
plt.title('Bedrooms\n', fontsize=16)

plt.axvline(np.median(df['bedrooms']), color='red', linestyle='dashed', linewidth=1.5, label='Médiane')
plt.axvline((df['bedrooms'].mean()), color='blue', linestyle='dotted', linewidth=1.5, label='Moyenne')

plt.legend()

plt.show()

File: task_2/hist.R
"""
Tâche 2:
2. Histogramme en R
"""

# Importons les bibliothèques nécessaires

library(ggplot2)
library(readr)

# Chargeons le dataset

df <- read_csv('Tache_2\\Housing_cleaned.csv')


# Réalisons l'histogramme

hist <- ggplot(df, aes(x=bedrooms)) +
  geom_histogram(binwidth = 0.5, fill='skyblue', color='black', alpha=0.7) +
  geom_vline(aes(xintercept = median(df$bedrooms)), color='red', linetype='dashed', linewidth=1) +
  geom_vline(aes(xintercept = mean(df$bedrooms)), color='blue', linetype='dotted', linewidth=1) +
  labs(
    title = 'Bedrooms\n',
    x = '\nNombre de chambres',
    y = 'Fréquences\n') +
  theme_minimal() +
  theme (
    plot.title = element_text (size = 18, hjust = 0.45, face = "bold"),
    axis.title.x = element_text (size = 14),
    axis.title.y = element_text (size = 14)
  )

print(hist)